examples/ui/drag_drop.rs

```diff
@@ -0,0 +1,374 @@
+use bevy::{
+    input::mouse::{MouseButtonInput, MouseMotion},
+    prelude::*,
+    render::texture::ImageSettings,
+    ui::FocusPolicy,
+};
+
+// Markers
+#[derive(Component)]
+struct Dragable;
+
+#[derive(Component)]
+struct DragActive;
+
+#[derive(Component)]
+struct DragMoving;
+
+#[derive(Component)]
+struct DragDropping;
+
+#[derive(Component)]
+struct DragStarting;
+
+#[derive(Component)]
+struct SourceDropContainer(pub Entity);
+
+#[derive(Component)]
+pub struct DropContainer(bool);
+
+impl Default for DropContainer {
+    fn default() -> Self {
+        DropContainer(true)
+    }
+}
+
+struct DragableHoverEvent(pub Entity);
```

Comment:
It's very strange to me that the fields are `pub` but the struct isn't. For an example I would make neither `pub`.

Reply:
I copied it out of my project where its seperated but its ugly indeed :)